### PR TITLE
feat: Made invoice similar to Eventbrite

### LINKED ISSUE #5891: Make Order Invoice similar to EventBrite
**Describe the bug**
Parent Issue:
https://github.com/fossasia/open-event-server/issues/5878

**Expected behavior**
The current order invoice needs to be generated as a proper PDF with the details specified here:
https://github.com/fossasia/open-event-server/pull/5886#issuecomment-492976538

_**While calculating the tax, the percentage and the amount must be shown**_

**Additional context**
Will work on this as soon as https://github.com/fossasia/open-event-server/pulls/5886 is merged.

### PR DESCRIPTION
**Pull request by mrsaicharan1**
Screenshot 2019-06-03 at 1 10 02 AM

Fixes #5891 

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

- [x] All the functions created/modified in this PR contain relevant docstrings.

#### Short description of what this resolves:
Structured the order invoice with order and related ticket info
#### Changes proposed in this pull request:

- Added VAT and VAT amount to invoice
- Added checks for billing info

### CODE REVIEW

**Comment by mrsaicharan1** [comment]
@shreyanshdwivedi @uds5501 Could you suggest some design changes to this invoice structure?

**Comment by uds5501** [comment]
@mrsaicharan1 I will pull the changes on my local and suggest edits in a while

**Comment by mrsaicharan1** [comment]
Sure

File: app/templates/pdf/order_invoice.html [hunk 1]
```diff
@@ -1,126 +1,228 @@
...
+                        <th scope="col">Quantity</th>
+                        <th scope="col">Sub-Total(net)</th>
+                        <th scope="col">VAT</th>
+                        <th scope="col">Tax Amount</th>
+                      </tr>
+                    </thead>
+                    <tbody>
+                    {% for ticket in tickets %}
+                      <tr>
+                        <td>{{ ticket.name }}</td>
```
Comment (uds5501) [inline app/templates/pdf/order_invoice.html]: Use custom alignment.
`<td style="text-align:center">` This style should match the `<thead>` tag's alignment which you are currently using

Comment (mrsaicharan1) [reply app/templates/pdf/order_invoice.html]: @uds5501 TYup.

**Comment by kushthedude** [comment]
@mrsaicharan1 Some suggestions for invoice -:

* The above two sub heading `Order` and `order details` can be replace with `Order Details` centre  aligned .
* Begin the alignment of `Order Details ` from extreme right margin .
* Try to have `Margins` for the tables and highlight the `Order Details Headings` as done in invoices generated by E-Com Sites .
* Ensure the following order of details in the `Order Details ` table -:
    * Ticket Type 
    * Price
    * Quantity
    * Discount (If Exist)
    * Net Amount
    * Tax Rate
    * Tax Amount
    * Total Amount

@uds5501 @shreyanshdwivedi @iamareebjamal Your views upon the following suggestions

**Comment by mrsaicharan1** [comment]
@mariobehling Can you please check the following screenshot for the order invoice? I would like to change it accordingly if we need anything else in this.

**Comment by mrsaicharan1** [comment]
> @mrsaicharan1 Some suggestions for invoice -:
> 
> * The above two sub heading Order and order details can be replace with Order Details centre aligned .
> * Begin the alignment of Order Details from extreme right margin .
> * Try to have Margins for the tables and highlight the Order Details Headings as done in invoices generated by E-Com Sites .
> * Ensure the following order of details in the Order Details table -:
>   
>   * Ticket Type
>   * Price
>   * Quantity
>   * Discount (If Exist)
>   * Net Amount
>   * Tax Rate
>   * Tax Amount
>   * Total Amount
> 
> @uds5501 @shreyanshdwivedi @iamareebjamal Your views upon the following suggestions

Keeping the headings bold is a way of highlighting it. The margins are not showing up for some weird reasons, even though I try to place them. I'll take care of the rest.

**Comment by mrsaicharan1** [comment]
@shreyanshdwivedi @uds5501 @iamareebjamal Please review. The order invoice looks like this now.
Screenshot 2019-06-03 at 1 10 02 AM

**Comment by uds5501** [comment]
@mrsaicharan1 how would it look without any billing information? in case the buyer chooses not to provide any billing info?

**Comment by mrsaicharan1** [comment]
> @mrsaicharan1 how would it look without any billing information? in case the buyer chooses not to provide any billing info?

Screenshot 2019-06-03 at 1 21 06 AM

**Comment by kushthedude** [comment]
@mrsaicharan1 Can you please show an invoice with Discount Amount ?

**Comment by kushthedude** [comment]
I still dont see `Net Amount` and `Price of Ticket ` in invoice .

**Comment by mrsaicharan1** [comment]
@kushthedude Please try to check clearly. Net amount is **Sub-total(net)**. Price of ticket is **price**

File: app/api/helpers/order.py [hunk 2]
```diff
@@ -70,7 +71,9 @@ def create_pdf_tickets_for_holder(order):
             save_to_db(holder)
 
         # create order invoices pdf
-        create_save_pdf(render_template('pdf/order_invoice.html', order=order, event=order.event),
+        order_ticket_info = OrderTicket.query.filter_by(order_id=order.id).one()
+        create_save_pdf(render_template('pdf/order_invoice.html', order=order, event=order.event,
+                                        tax=order.event.tax, tickets=order.tickets, order_tickets_info=order_ticket_info),
```
[the remark this answers is not in the document]
Comment (uds5501) [reply app/api/helpers/order.py]: @mrsaicharan1 kindly fix this, otherwise it looks GTG

**Comment by kushthedude** [comment]
> @kushthedude Please try to check clearly. Net amount is **Sub-total(net)**. Price of ticket is **price**

Can you shift that column to the last and shift tax columns before Net

**Comment by mrsaicharan1** [comment]
@uds5501 Hound issues fixed. :)

**Comment by mrsaicharan1** [comment]
Screenshot 2019-06-03 at 1 31 04 AM

EventBrite follows this format. Therefore, I wouldn't prefer to change it.

**Pull request merged**